spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,220 @@
+            raise ValueError("n_max must be a positive integer")
+        if not data.shape[-2] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few sh coefficients for '
+                             f'{n_max=}. Highest possible n_max is '
+                             f'{int(np.sqrt(data.shape[-2]))-1}')
+        self._n_max = int(n_max)
+
+        # set basis_type
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
+
+        if basis_type not in ["complex", "real"]:
+            raise ValueError("Invalid basis type, only "
+                             "'complex' and 'real' are supported")
+        self._basis_type = basis_type
+
+        # set normalization
+        if normalization not in ["n3d", "maxN", "sn3d"]:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN', but is {normalization}")
+        self._normalization = normalization
+
+        # set channel_convention
+        if channel_convention not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {channel_convention}")
+        self._channel_convention = channel_convention
+
+        # set condon shortley
+        if not isinstance(condon_shortley, bool) and condon_shortley != 'auto':
+            raise ValueError(
+                "Condon_shortley has to be a bool, or 'auto'.")
+
+        if condon_shortley == 'auto' and basis_type == 'complex':
+            self._condon_shortley = True
+        elif condon_shortley == 'auto' and basis_type == 'real':
+            self._condon_shortley = False
+
+        Signal.__init__(self, data, sampling_rate=sampling_rate,
+                        n_samples=n_samples, domain=domain, fft_norm=fft_norm,
+                        comment=comment, is_complex=is_complex)
+
+    @property
+    def n_max(self):
+        return self._n_max
+
+    @property
+    def basis_type(self):
+        return self._basis_type
+
+    @property
+    def normalization(self):
+        return self._normalization
+
+    @normalization.setter
+    def normalization(self, value):
+        if self.normalization is not value:
+            self._renormalize(value)
+
+    @property
+    def channel_convention(self):
+        return self._channel_convention
+
+    @property
+    def condon_shortley(self):
+        return self._condon_shortley
+
+    @channel_convention.setter
+    def channel_convention(self, value):
+        if value not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {value}")
+
+        if self.channel_convention is not value:
+            self._change_channel_convention()
+
+    def _renormalize(self, value):
```

Comment:
This will also be something that needs to be done in the SH class. We should think about the best way to share this code. Probably by making it a function.